docs/doxygen/other/mimo.dox

```diff
@@ -0,0 +1,420 @@
+This leads to the following combining algorithm (if there are no equal csi coefficients):
+
+\f[
+\alpha_i = 1, \quad \text{for} \quad \text{csi}[i] > \text{csi}[j] \quad \text{and} \quad  j \neq i, \quad 0 \quad \text{otherwise}
+\f]
+
+The selection combining mitigates the effects of fading by constantly switching 
+to the path with the least attenuation. It has obviously no array gain because 
+it only takes one received signal at each time into account.
+
+\subsection mimo_maximum_ratio_combining Maximum-Ratio Combining (MRC)
+
+Optimizing the weighted sum in favor of a maximum SNR results in maximum-ratio combining:
+
+\f[
+\alpha_i = \frac{\text{csi}[i]}{E} \quad \text{with} \quad E = \sum_{i=1}^N \text{csi}_i^2
+\f]
+
+The magnitude weighting of each path is proportional to its path coefficient. 
+Afterwards it is normalized to avoid an amplification of the signal. 
+The phase of the weighting elements is the inverse of the phase shift of its related path coefficient. 
+Therefore, all products of the weighted sum have the same phase.  
+This so called ‘co-phasing‘ [1] leads to a constructive interference of the 
+different path signals instead of a random interference that could also lead 
+to a signal cancellation in the case of destructive interference, 
+which would obviously not maximize the SNR.
+
+\subsection mimo_diversity_combining_usage Usage
+
+The diversity combiner is implemented in the C++ block gr::digital::diversity_combiner_cc. 
+It takes N complex input ports and has one complex output port. The combining 
+technique is stored as a string type in a local variable. This has the advantage, 
+that it can be changed during run-time with a respective setter method. 
+The block is generally working in stream mode but vector passing can be enabled by setting the vector length > 1.
+
+CSI is provided via stream tags with the key ‘csi’. With each incoming CSI stream tag, 
+the weighting vector is calculated with the new data. The idea is, 
+that a stream tag comes with each new symbol, where symbol means a frame of data samples which belong together. 
+The definition of a symbol (and therefore the frequency of tagging CSI) is the job of the system designer who uses this block.
+
+\subsection mimo_diversity_combining_csi Providing CSI
+
+Providing channel state information (CSI) via stream tags is a fundamental 
+requirement for the diversity combining block and also for many other MIMO blocks 
+which are introduced later on. The channel can be estimated at the receiver with pilot symbols, for example. 
+A detailled instruction to channel estimation for MIMO channels is not part of 
+this documentation but you can have an inpirational look at the channel estimation 
+of the MIMO-OFDM transceiver [not yet implemented; scheduled for August 2018].
+
+The produced CSI is a channel matrix of the dimensions NxM. In case of diversity combining, 
+the matrix reduces to a N-dimensional vector.
```

Comment:
Why N-dimensional? I would have expected that after diversity combining, it's M coefficients (for the different transmit antennas). Probably I'm confusing things here, but maybe clarify this.